packages/auth/src/platform_browser/index.ts

```diff
@@ -103,4 +106,32 @@ export function getAuth(app: FirebaseApp = getApp()): Auth {
   return auth;
 }
 
+function getScriptParentElement(): HTMLDocument | HTMLHeadElement {
+  return document.getElementsByTagName('head')?.[0] ?? document;
+}
+
+_setExternalJSProvider({
+  loadJS(url: string): Promise<Event> {
+    // TODO: consider adding timeout support & cancellation
+    return new Promise((resolve, reject) => {
+      const el = document.createElement('script');
+      el.setAttribute('src', url);
+      el.onload = resolve;
+      el.onerror = e => {
+        const error = _createError(AuthErrorCode.INTERNAL_ERROR);
+        error.customData = e as unknown as Record<string, unknown>;
+        reject(error);
+      };
+      el.type = 'text/javascript';
+      el.charset = 'UTF-8';
+      getScriptParentElement().appendChild(el);
+    });
+  },
+
+  gapiScript: 'https://apis.google.com/js/api.js',
+  recaptchaV2Script: 'https://www.google.com/recaptcha/api.js',
+  recaptchaEnterpriseScript:
+    'https://www.google.com/recaptcha/enterprise.js?render='
```

Comment:
Is `?render=` needed?

Reply:
Yes, I copied the exact url from https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts#L35.

The full url we need to [pass in _loadJS](https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts#L138) is 'https://www.google.com/recaptcha/enterprise.js?render=SITE-KEY'.

So if we remove '?render=' from here, we will have to update [this line](https://github.com/firebase/firebase-js-sdk/pull/7766/files/9fcd5d82d942373798adedb5d0e9c390bbaff24b#diff-f6ec4be48b8bf6e28bb2a3915b3c5a43c97e6749c5405099af825ca13f031ec7R136) to be
```
url += '?render=' + siteKey;
```